packet_logger.go

```diff
@@ -0,0 +1,73 @@
+package glutton
+
+import (
+	"fmt"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"github.com/google/gopacket/pcapgo"
+	"os"
+	"time"
+)
+
+var (
+	deviceName   string = "wlp3s0"
+	snapshotLen  int32  = 1024
+	snapshot_len uint32 = 1024
+	promiscuous  bool   = false
+	err          error
+	timeout      time.Duration = -1 * time.Second
+	handle       *pcap.Handle
+)
+
+func StartCapturing() {
+	// Open output pcap file and write header
+	name := "/var/log/glutton/" + time.Now().String() + ".pcap"
+	f, err := os.OpenFile(name, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600)
+	if err != nil {
+		println("[*] Error in creating pcap file", err.Error())
+		return
+	}
+	defer f.Close()
+
+	w := pcapgo.NewWriter(f)
+	w.WriteFileHeader(snapshot_len, layers.LinkTypeEthernet)
+
+	// Open the device for capturing
+	handle, err = pcap.OpenLive(deviceName, snapshotLen, promiscuous, timeout)
+	if err != nil {
+		fmt.Printf("Error opening device %s: %v", deviceName, err)
+		return
+	}
+	defer handle.Close()
+	err = handle.SetBPFFilter("tcp or udp")
+	if err != nil {
+		println("[*] Error: BPF filtering", err.Error())
+	}
+
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+	for packet := range packetSource.Packets() {
+		w.WritePacket(packet.Metadata().CaptureInfo, packet.Data())
+	}
+
+}
+
+func scanForDevices() {
```

Comment:
Yes. In case if machine have one interface glutton will automatically select that one. Otherwise glutton will look into configuration.